src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -691,8 +691,9 @@ def set_functionapp(cmd, resource_group_name, name, **kwargs):
 
 
 def list_webapp(cmd, resource_group_name=None):
-    result = _list_app(cmd.cli_ctx, resource_group_name)
-    return [r for r in result if 'function' not in r.kind]
+    full_list = _list_app(cmd.cli_ctx, resource_group_name)
+    # ignore apps with kind==null & not functions apps
+    return list(filter(lambda x: x.kind is not None and "function" not in x.kind.lower(), full_list))
```

Comment:
Is it possible that full_list is None not []?

Reply:
the API returns a array so it is [] & not None.